Products/ZenEvents/tests/test_zentrap.py

```diff
@@ -70,8 +76,670 @@ def test_decode_unexpected_object_type(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1Enterprise"], "1.2.3.4")
+
+    def test_SnmpV1GenericTrapType(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1GenericTrapType"], 6)
+
+    def test_SnmpV1SpecificTrap(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1SpecificTrap"], 5)
+
+    def test_EnterpriseOIDWithoutExtraZero(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "1.2.3.4.5")
+        self.assertEqual(result["oid"], "1.2.3.4.5")
+
+    def test_EnterpriseOIDWithExtraZero(self):
+        pckt = self.makePacket()
+        oidMap = {"1.2.3.4.0.5": "testing"}
+        task = MockTrapTask(oidMap)
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "testing")
+        self.assertEqual(result["oid"], "1.2.3.4.0.5")
+
+    def test_TrapTypeDecoding(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+
+        pckt.trap_type = 0
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "coldStart")
+        self.assertEqual(result["snmpV1GenericTrapType"], 0)
+
+        pckt.trap_type = 1
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "warmStart")
+        self.assertEqual(result["snmpV1GenericTrapType"], 1)
+
+        pckt.trap_type = 2
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "snmp_linkDown")
+        self.assertEqual(result["snmpV1GenericTrapType"], 2)
+
+        pckt.trap_type = 3
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "snmp_linkUp")
+        self.assertEqual(result["snmpV1GenericTrapType"], 3)
+
+        pckt.trap_type = 4
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "authenticationFailure")
+        self.assertEqual(result["snmpV1GenericTrapType"], 4)
+
+        pckt.trap_type = 5
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "egpNeighorLoss")
+        self.assertEqual(result["snmpV1GenericTrapType"], 5)
+
+        pckt.trap_type = 6
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "1.2.3.4.5")
+        self.assertEqual(result["snmpV1GenericTrapType"], 6)
```

Comment:
I would break these up into separate tests, since they are testing decoding on different tryap types.

You can deduplicate the  ```pckt = self.makePacket() \ task = MockTrapTask({})``` boilerplate by adding it to a setUp method for the test Class.  like so:

```
def setUp(self):
    self.pckt = self.makePacket()
    self.task = MockTrapTask({})
```